MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2443,7 +2443,13 @@ public string GetHardwareAcceleratedVideoDecoder(VideoType videoType, MediaStrea
                     {
                         case "avc":
                         case "h264":
-                            if (_mediaEncoder.SupportsDecoder("h264_cuvid") && encodingOptions.HardwareDecodingCodecs.Contains("h264", StringComparer.OrdinalIgnoreCase))
+                            // cuvid decoder does not support >8 bits on h264, fall back onto h264.
+                            if (videoStream.BitDepth.HasValue && videoStream.BitDepth.Value > 8)
```

Comment:
This check should be made within the check that is seeing whether hw acceleration is requested